sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java

```diff
@@ -284,19 +297,41 @@ public Object getPartitionKeyFieldValue(T entity) {
                 }
             });
             return currentObject[0];
+        } else if (hierarchicalPartitionKeyPaths != null && hierarchicalPartitionKeyPaths.length > 0) {
+            ArrayList<Object> pkValues = new ArrayList<>();
+            for (final String path : hierarchicalPartitionKeyPaths) {
+                Field f = null;
+                try {
+                    f = entity.getClass().getDeclaredField(path.substring(1));
```

Comment:
I think even for hierarchicalPartitionKeys, each path can also be nested partitionKey, so we might need to do the same parsing as line 285 - 298

Reply:
@kushagraThapar do we allow nested PK's in hierarchical PK's? If so we will need to figure out our approach.

Nested PK's are defined like this:
`@Container(partitionKeyPath = "/nestedEntitySample/nestedPartitionKey")`

And Hierarchical PK's are defined like this:
`Container(hierarchicalPartitionKeyPaths = {"/id", "/firstName", "/lastName"})`

The only way we currently know what type of PK it is is from the name identifier, we don't determine it based on the structure of the text. With this there is no way for us to know if there is a nested key in a hierarchical key. Thoughts?

Reply:
Hmm, @xinlian12 can you please explain with some example?

Reply:
I do not know whether it is a valid scenario, I am wondering whether the following config is valid:
 Container(hierarchicalPartitionKeyPaths = {"/id", "/nestedEntity/firstName", "/nestedEntity/lastName"})

Reply:
Agreed, I think we should test it out - @trande4884 - can we test this out and see if it passes or fails. One way to test this would be first directly in the Java SDK and then in spring data cosmos. Given this is the last open comment, let's try to get some clarity on it and then we should be able to release this work :)

Reply:
Unless we fundamentally change how we handle PK's that example is not going to work. We have no way of determining that a key that is just a string in the heirarchical key is a nested key, the only way we know if a key is nested is if it defined by the variable "partitionKeyPath" like the following:
@Container(containerName = "nested-partition-key", partitionKeyPath = "/nestedEntity/nestedPartitionKey",
    ru = TestConstants.DEFAULT_MINIMUM_RU)
    
If you look at CosmosEntityInformation in this PR you will see we don't dynamically identify what kind of type of PK it is, we either look for one established with the @PartitionKey tag (standard PK) or one to be setup with the identifier "partitionKeyPath" (nested PK) or "hierarchicalPartitionKeyPaths" (sub PK). If we want to fundamentally change how we do this to dynamically identify the PK's and the corresponding type I am not opposed but I believe it should be done outside of this PR especially as it would be a breaking change.

Reply:
Makes sense @trande4884 - and we don't want to put a breaking change right now. So may be we will leave it as it is for now, and if its a limitation, let it be for at least the first couple of releases with sub-partitioning.